src/Modules/Math/BooleanCompare.cc

```diff
@@ -0,0 +1,110 @@
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+   DEALINGS IN THE SOFTWARE.
+*/
+
+#include <Modules/Math/BooleanCompare.h>
+#include <Core/Datatypes/Matrix.h>
+#include <Core/Datatypes/DenseMatrix.h>
+#include <Core/Algorithms/Base/AlgorithmVariableNames.h>
+#include <Dataflow/Network/Module.h>
+#include <Core/Datatypes/MatrixTypeConversions.h>
+#include <Core/Algorithms/Math/BooleanCompareAlgo.h>
+#include <Dataflow/Engine/Python/NetworkEditorPythonAPI.h>
+
+using namespace SCIRun::Modules::Math;
+using namespace SCIRun::Dataflow::Networks;
+using namespace SCIRun::Core::Algorithms;
+using namespace SCIRun::Core::Algorithms::Math;
+using namespace SCIRun::Core::Datatypes;
+
+/// @class BooleanCompare
+/// This module is for comparing two matrices and sending outputs
+
+MODULE_INFO_DEF(BooleanCompare, Flow Control, SCIRun)
+
+BooleanCompare::BooleanCompare() : Module(staticInfo_)
+{
+  INITIALIZE_PORT(MatrixA);
+  INITIALIZE_PORT(MatrixB);
+  INITIALIZE_PORT(PossibleOutput);
+  INITIALIZE_PORT(OutputMatrix);
+  INITIALIZE_PORT(BooleanResult);
+}
+
+void BooleanCompare::setStateDefaults()
+{
+  setStateStringFromAlgoOption(Parameters::Comparison_Option);
+  setStateStringFromAlgoOption(Parameters::Value_Option_1);
+  setStateStringFromAlgoOption(Parameters::Value_Option_2);
+  setStateStringFromAlgoOption(Parameters::Then_Option);
+  setStateStringFromAlgoOption(Parameters::Else_Option);
+}
+
+void
+BooleanCompare::execute()
+{
+  auto matrixA = getRequiredInput(MatrixA);
+  auto matrixB = getOptionalInput(MatrixB);
+  auto possout = getOptionalInput(PossibleOutput);
+
+  if (needToExecute())
+  {
+      
+    setAlgoOptionFromState(Parameters::Comparison_Option);
+    setAlgoOptionFromState(Parameters::Value_Option_1);
+    setAlgoOptionFromState(Parameters::Value_Option_2);
+    setAlgoOptionFromState(Parameters::Then_Option);
+    setAlgoOptionFromState(Parameters::Else_Option);
+    
+
+    auto output = algo().run(withInputData((Variables::FirstMatrix, matrixA)(Variables::SecondMatrix, optionalAlgoInput(matrixB))(Variables::InputMatrix, optionalAlgoInput(possout))));
+    
+//    Check for quit condition and then quit.  Arguably the most useful part of the module
+    auto state = get_state();
+    std::string then_statement = state->getValue(Parameters::Then_Option).toString();
+    std::string else_statement = state->getValue(Parameters::Else_Option).toString();
+    
+    auto out_mat=output.get<DenseMatrix>(Variables::Solution);
+    double *data=out_mat->data();
+    
+    if ((data[0]==1 && then_statement == "quit") || (data[0]==0 && else_statement == "quit"))
+    {
+
+      std::cout<<"cond matrix :"<<*out_mat<<std::endl;
+      std::cout<<"ordered to quit.  Trying to quit."<<std::endl;
+//      quit nicely
+      std::string q_message=NetworkEditorPythonAPI::quit(false);
```

Comment:
This part forces BUILD_WITH_PYTHON to be enabled. To fix the gcc build (which is run without python enabled), you can make your entire module require python, or just this section. 

If the former, toggle including the file in the CMakeLists and surround the whole file in `#ifdef BUILD_WITH_PYTHON`. If the latter, just surround the quit code with the same macro.